lib/msal-common/src/request/AuthenticationHeaderParser.ts

```diff
@@ -54,20 +55,40 @@ export class AuthenticationHeaderParser {
         throw ClientConfigurationError.createMissingNonceAuthenticationHeadersError();
     }
 
+    /**
+     * This method parses the claims value out of the WWW-Authenticate authentication header.
+     * See: https://docs.microsoft.com/en-us/azure/active-directory/develop/claims-challenge
+     * @returns 
+     */
+    getClaims(): string {
+        // Attempt to parse claims from WWW-Authenticate
+        const wwwAuthenticate = this.headers[HeaderNames.WWWAuthenticate];
+        if (wwwAuthenticate) {
+            const wwwAuthenticateChallenges = this.parseChallenges<WWWAuthenticateChallenges>(wwwAuthenticate);     
```

Comment:
As per spec, there can be multiple WWW-Authenticate headers. `getShrNonce` should really only look at the WWW-Authenticate header with auth_scheme `POP` (it currently looks at any scheme). 

For claims however, it's trickier. There could be multiple claims parameters associated with various auth schemes. Getting the "first" one seems wrong. 

```
WWW-Authenticate: Bearer authority=https://lmo.com/123 claims=some_claims
WWW-Authenticate: POP authority=https://lmo.com/123 claims=some_other_claims nonce=n123
```

The spec @trwalke wrote has details on how to treat this case.

Reply:
This seems to be [the case](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/WWW-Authenticate#syntax). Currently we are treating the header as key-value pairs, and JS does not allow duplicate keys, so the last one overrides. I'll convert this to draft for now since this seems to need a redesign. cc @hectormmg

Reply:
Interesting, I wasn't aware there could be multiple headers with the same name. I agree we'll need to update this to handle multiple headers.